main.go

```diff
@@ -306,18 +329,46 @@ func PumpIt(enumerator pump.Enumerator, collector pump.Collector, drain pump.Dra
 		go func() {
 			for block := range blocks {
 				if block.Error != nil {
-					log.Println(errors.Wrapf(block.Error, "error retrieving bloc %s", block.CID.String()))
+					log.Println(errors.Wrapf(block.Error, "error retrieving block %s", block.CID.String()))
+					failedBlocks <- block.CID
 					continue
 				}
 
 				err = drain.Drain(block)
 				if err != nil {
 					log.Println(errors.Wrapf(err, "failed to push block %s", block.CID.String()))
+					failedBlocks <- block.CID
+					continue
 				}
 			}
 			wgDrain.Done()
 		}()
 	}
 
-	wgDrain.Wait()
+	// Spawn 1 failed blocks writer worker (is enough)
```

Comment:
Since you only have one writer, why not drop entirely this channel/concurrency structure and just write to it directly? If there is concurrency issue, a simple mutex would fix that easily.

Reply:
Or even an `atomic` as you only have a counter.

Reply:
I wanted to avoid writing from multiple go-routines to the file, so I used this new channel to, pun intended, channel the CIDs into this one writer go-routine. It also follows the structure in this file so it will be easier in this way to scale up the workers handling the errors file if needed later. You know what I mean?

Reply:
if you add more workers to process the errors you end up with concurrent access on your writer so you are back to square one: you need to have your writer thread-safe. This complex mechanism here doesn't help you.

Reply:
If we add more workers, we need to only make the writer thread-safe. So not really a square one bc at least the scaling logic through channels is in place. My goal with this implementation was to do things exactly like everything else a few lines above (we do this pattern 3x in this file) to don't diverge from the codebase. It's your tool and it worked very well so let me know how you prefer it and I will adjust it.